src/core/layertree/qgslayertreemodellegendnode.cpp

```diff
@@ -539,7 +546,101 @@ void QgsSymbolLegendNode::updateLabel()
   emit dataChanged();
 }
 
+QString QgsSymbolLegendNode::evaluateLabel( QgsExpressionContext context, QString label )
+{
+  if ( !mLayerNode )
+    return QString();
+
+  QgsVectorLayer *vl = qobject_cast<QgsVectorLayer *>( mLayerNode->layer() );
+
+  if ( label.isEmpty() )
+  {
+    if ( mEmbeddedInParent )
+    {
+      QString layerName = mLayerNode->name();
+      if ( !mLayerNode->customProperty( QStringLiteral( "legend/title-label" ) ).isNull() )
+        layerName = mLayerNode->customProperty( QStringLiteral( "legend/title-label" ) ).toString();
+
+      mLabel = mUserLabel.isEmpty() ? layerName : mUserLabel;
+      if ( vl && ( !( mLayerNode->expression().isEmpty() ) || mLabel.contains( "[%" ) ) )
+      {
+        mLabel = evaluateLabelExpression( mLabel, vl, context );
+      }
+    }
+    else
+    {
+      mLabel = mUserLabel.isEmpty() ? mItem.label() : mUserLabel;
+      if ( vl && ( !( mLayerNode->expression().isEmpty() ) || mLabel.contains( "[%" ) ) )
+      {
+        mLabel = evaluateLabelExpression( mLabel, vl, context );
+      }
+    }
+    emit dataChanged();
+    return mLabel;
+  }
+  else
+  {
+    if ( vl && ( !( mLayerNode->expression().isEmpty() ) || label.contains( "[%" ) ) )
+    {
+      label = evaluateLabelExpression( label, vl, context );
+    }
+    return label;
+  }
+}
+
+QgsExpressionContext QgsSymbolLegendNode::createExpressionContext( QgsExpressionContext context ) const
+{
+  //= QgsLayoutItem::createExpressionContext();
+
+  QgsVectorLayer *vl = qobject_cast<QgsVectorLayer *>( mLayerNode->layer() );
+
+  // TODO: ADD NECESSARY SCOPES ??
+  context.appendScope( vl->createExpressionContextScope() );
```

Comment:
https://qgis.org/api/classQgsExpressionContextUtils.html#ae204a266b5a9ae5475f568c264fc719c

Reply:
I left the TODO for me in case I needed more scopes but I think that by passing down the layout legend+layout map and the context from the vector layer should be sufficient.

I'll remove the todo.